example-utils/usubscription-static-file/src/lib.rs

```diff
@@ -0,0 +1,184 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+#![allow(clippy::mutable_key_type)]
+
+use async_trait::async_trait;
+use serde_json::Value;
+use std::collections::HashSet;
+use std::fs::{self, canonicalize};
+use std::path::PathBuf;
+use std::str::FromStr;
+use up_rust::core::usubscription::{
+    FetchSubscribersRequest, FetchSubscribersResponse, FetchSubscriptionsRequest,
+    FetchSubscriptionsResponse, NotificationsRequest, SubscriberInfo, Subscription,
+    SubscriptionRequest, SubscriptionResponse, USubscription, UnsubscribeRequest,
+};
+use up_rust::{UStatus, UUri};
+
+pub struct USubscriptionStaticFile {
+    static_file: PathBuf,
+}
+
+impl USubscriptionStaticFile {
+    pub fn new(static_file: Option<PathBuf>) -> Self {
```

Comment:
I think it's better to force explicitness. IMHO it should not be `Option<PathBuf>`, but just `PathBuf`.

Reply:
okay changed that, take a look